code-challenges.js

```diff
@@ -6,70 +6,255 @@
 // If you get stuck, please leave comments to help us understand your thought process
 
+// DONT FORGET TO COMMENT ON WHAT EACH LINE OF CODE IS DOING ---------------------------------------------
 // --------------------1) Create a function that determines if a temperature is below boiling point, at boiling point or above boiling point. Boiling point is 212 degrees Fahrenheit.
 // Use the test variables provided below. Expected outcome: "85 is below boiling point" "350 is above boiling point" "212 is at boiling point"
 
 var temp1 = 85
 var temp2 = 350
 var temp3 = 212
 
+const tempCheck = (num) => {
 
+    let boil = 212
+    // set a variable to the boil value for simplicity
+    if(typeof num !== "number"){
+        return `${num} is not a number`
+        // check if the user inputs something other than a number
+    }
 
+    if(num < boil){
+        return `${num} is below boiling point.`
+        // if the number is below the value, print this ^
+    } else if(num > boil){
+        return `${num} is above boiling point.`
+        // if the number is above the value, print this ^
+    } else if(num === boil) {
+        return `${num} is at boiling point.`
+        // if the number is =to the value, print this ^
+    } else {
+        return "You are out of range"
+        // if you type in something other than a number, print this ^ (another bug check)
+    }
+}
+console.log(tempCheck(212))
 
 
-// --------------------2) Create a function that multiplies each number in the array by 5 using a for loop.
-// Use the test variable provided below. Expected outcome: [15, 35, 0, 30, -45] [10, 25, -40, 100, 160]
+
+// // --------------------2) Create a function that multiplies each number in the array by 5 using a for loop.
+// // Use the test variable provided below. Expected outcome: [15, 35, 0, 30, -45] [10, 25, -40, 100, 160]
 
 var myForNumbers1 = [3, 7, 0, 6, -9]
 var myForNumbers2 = [2, 5, -8, 20, 32]
 
+const multiply = (array) => {
+
+    let newArr = []
+    // initializing an empty array
 
+    for(i = 0; i < array.length; i++){
+        // start of for loop
+        newArr.push(array[i] * 5)
+        // adding every element to the empty newArr array multiplied by 5
+    }
+    return newArr
+    // return the new array
+} 
+console.log(multiply(myForNumbers1))
+console.log(multiply(myForNumbers2))
 
 
 
-// --------------------3) Create a function that multiplies each number in the array by 5 using map.
-// Use the test variable provided below. Expected outcome: [435, 30, 140, 80, 10] [40, -35, 0, 30, 60]
+// // --------------------3) Create a function that multiplies each number in the array by 5 using map.
+// // Use the test variable provided below. Expected outcome: [435, 30, 140, 80, 10] [40, -35, 0, 30, 60]
 
 var myMapNumbers1 = [87, 6, 28, 16, 2]
 var myMapNumbers2 = [8, -7, 0, 6, 12]
 
+const mapMult = (array) => {
+    // start of a higher order function taking in an array for its argument
+    return array.map(value => {
```

Comment:
for simple return statements you can use an implicit return 
```suggestion
    return array.map(value => value*5)
```